pkg/olm/catalog.go

```diff
@@ -0,0 +1,309 @@
+	var csvMetadata *property.CSVMetadata
+	for _, p := range bundle.Properties {
+		if p.Type == property.TypeCSVMetadata {
+			if err := json.Unmarshal(p.Value, &csvMetadata); err != nil {
+				return nil, fmt.Errorf("failed to unmarshal csv metadata for bundle %q: %w", bundle.Name, err)
+			}
+		}
+	}
+	return csvMetadata, nil
+}
+
+func parseJSONArray(arr string) []string {
+	var parsed []string
+	if err := json.Unmarshal([]byte(arr), &parsed); err != nil {
+		return nil
+	}
+	return parsed
+}
+
+func parseCommaSeparatedString(val string) []string {
+	if val == "" {
+		return []string{}
+	}
+	parts := strings.Split(val, ",")
+	result := []string{}
+	for _, part := range parts {
+		part = strings.TrimSpace(part)
+		if part != "" {
+			result = append(result, part)
+		}
+	}
+	return result
+}
+
+func getInfrastructureFeatures(csvMetadata *property.CSVMetadata) []string {
+	infrastructureFeatures := []string{}
+	if csvMetadata.Annotations[InfrastructureFeaturesOLMAnnotationKey] != "" {
+		infrastructureFeatures = parseJSONArray(csvMetadata.Annotations[InfrastructureFeaturesOLMAnnotationKey])
+	}
+
+	infrastructureFeatureSet := make(map[string]bool)
+	for _, feature := range infrastructureFeatures {
+		infrastructureFeatureSet[feature] = true
+	}
+
+	for _, annotationKey := range infrastructureFeatureAnnotations {
+		if csvMetadata.Annotations[annotationKey] == "true" {
+			infrastructureFeatureSet[annotationKey] = true
+		}
+	}
+
+	infrastructureFeatures = []string{}
+	for feature := range infrastructureFeatureSet {
+		infrastructureFeatures = append(infrastructureFeatures, feature)
+	}
+	sort.Strings(infrastructureFeatures)
+	return infrastructureFeatures
+}
+
+func withCapabilities(item *ConsoleCatalogItem, csvMetadata *property.CSVMetadata) {
+	if csvMetadata == nil {
+		return
+	}
+
+	capabilities := csvMetadata.Annotations[CapabilitiesOLMAnnotationKey]
+	if capabilities != "" {
+		item.Capabilities = capabilities
+	}
+}
+
+func withCategories(item *ConsoleCatalogItem, csvMetadata *property.CSVMetadata) {
+	if csvMetadata == nil {
+		return
+	}
+
+	categories := parseCommaSeparatedString(csvMetadata.Annotations[CategoriesOLMAnnotationKey])
+	if len(categories) > 0 {
+		item.Categories = categories
+	}
+}
+
+func withCreatedAt(item *ConsoleCatalogItem, csvMetadata *property.CSVMetadata) {
+	if csvMetadata == nil {
+		return
+	}
+
+	createdAt := csvMetadata.Annotations[CreatedAtOLMAnnotationKey]
+	if createdAt != "" {
+		item.CreatedAt = createdAt
+	}
+}
+
+func withDescription(item *ConsoleCatalogItem, csvMetadata *property.CSVMetadata, pkg *declcfg.Package) {
+	item.Description = pkg.Description
+	if csvMetadata == nil {
+		return
+	}
+
+	if csvMetadata.Description != "" {
+		item.Description = csvMetadata.Description
+	}
+
+	if csvMetadata.Annotations[DescriptionOLMAnnotationKey] != "" {
+		item.Description = csvMetadata.Annotations[DescriptionOLMAnnotationKey]
+	}
+
+	item.Description = pkg.Description
```

Comment:
Why we need to set `item.Description` to `pkg.Description` again? Won't that overwrites everything (e.g `tem.Description = csvMetadata.Description`) above?